### PR TITLE
[FIX] *: remove the payment_demo dependency and install when user ticks import demo data

### PR DESCRIPTION
**Pull request by dhrs-odoo**
To remove the module from the dependency, and to install it only if the user ticks the import demo data checkbox. 

Task-4018764

### CODE REVIEW

File: agriculture_shop/__manifest__.py [hunk 1]
```diff
@@ -12,7 +12,6 @@
         'pos_sale',
         'product_expiry',
         'purchase_requisition',
-        'payment_demo',
```
Comment (vava-odoo) [inline agriculture_shop/__manifest__.py:15]: Could you add a test that ensures payment_demo is not added to the dependency list anymore?

File: agriculture_shop/data/payment_provider.xml [hunk 2]
```diff
@@ -1,7 +0,0 @@
-<?xml version='1.0' encoding='UTF-8'?>
-<odoo noupdate="1">
-    <record id="payment.payment_provider_demo" model="payment.provider">
-        <field name="is_published" eval="0"/>
-        <field name="state">disabled</field>
-    </record>
```
Comment (vava-odoo) [inline agriculture_shop/data/payment_provider.xml:6]: Could you remove the test checking these files?

File: agriculture_shop/demo/payment_provider_demo_post.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
```
Comment (vava-odoo) [inline agriculture_shop/demo/payment_provider_demo_post.xml:2]: Probably better in noupdate, since we don't want to force the installation everytime

File: fmcg_store/__manifest__.py [hunk 4]
```diff
@@ -55,8 +53,7 @@
         'demo/purchase_order_post.xml',
         'demo/sale_order_post.xml',
         'demo/pos_payment_method.xml',
-        'demo/payment_provider.xml',
-        'demo/payment_method.xml',
```
Comment (vava-odoo) [inline fmcg_store/__manifest__.py:59]: Why did you remove this file?

File: fmcg_store/demo/payment_provider.xml [hunk 5]
```diff
@@ -1,15 +0,0 @@
...
-<odoo noupdate="1">
-    <record id="payment.payment_provider_demo" model="payment.provider">
-        <field name="name">Card Payment</field>
-        <field name="payment_method_ids" eval="[
-            Command.link(ref('payment.payment_method_card')),
-            Command.link(ref('payment.payment_method_visa')),
-            Command.link(ref('payment.payment_method_mastercard')),
-            Command.link(ref('payment.payment_method_rupay')),
-            Command.link(ref('payment.payment_method_maestro')),
-        ]"/>
```
Comment (vava-odoo) [inline fmcg_store/demo/payment_provider.xml:11]: This should probably be kept, no?

File: pharmacy_retail/__manifest__.py [hunk 6]
```diff
@@ -65,8 +63,7 @@
         'demo/loyalty_generate_wizard.xml',
         'demo/sale_loyalty_wizard.xml',
         'demo/sale_order_post.xml',
-        'demo/payment_provider.xml',
-        'demo/payment_method.xml',
```
Comment (vava-odoo) [inline pharmacy_retail/__manifest__.py:69]: Same comment

**Review by vava-odoo — COMMENTED** [review]
Thanks @dhrs-odoo 
It looks good and is way cleaner.
I had a few comments, could you have a look?
Cheers!

**Comment by dhrs-odoo** [comment]
Hello @vava-odoo 
I have incorporated the changes 
Please have a look at it once you are free
Thank You :innocent:

File: pharmacy_retail/demo/payment_provider.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="payment.payment_provider_demo" model="payment.provider">
+        <field name="name">Card Payment</field>
+        <field name="is_published" eval="1"/>
+        <field name="state">test</field>
```
Comment (vava-odoo) [inline pharmacy_retail/demo/payment_provider.xml:6]: These can be removed (default)
```suggestion
```

**Review by vava-odoo — APPROVED** [review]
Thanks @dhrs-odoo!
